### PR TITLE
Weighted updates for kll items sketch

### PR DESCRIPTION
**Pull request by leerho**

### CODE REVIEW

File: src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java [hunk 1]
```diff
@@ -205,13 +221,64 @@ public void viewCompactionAndSortedView() {
...
+      assertEquals(wt, correct[i++]);
     }
   }
 
+  @Test //set static enablePrinting = true for visual checking
+  public void checkWeightedUpdates1() {
+    int k = 20;
+    int weight = 127;
+    String item = "10";
+    KllItemsSketch<String> sk = KllItemsSketch.newHeapInstance(20, Comparator.naturalOrder(), serDe);
```
Comment (AlexanderSaydakov) [inline src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java:239]: should be k, which is defined above and never used

File: src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java [hunk 2]
```diff
@@ -205,13 +221,64 @@ public void viewCompactionAndSortedView() {
...
+  }
+
+  @Test //set static enablePrinting = true for visual checking
+  public void checkWeightedUpdates2() {
+    int k = 20;
+    int initial = 1000;
+    final int digits = 4;
+    int weight = 127;
+    String item = "  10";
+    KllItemsSketch<String> sk = KllItemsSketch.newHeapInstance(20, Comparator.naturalOrder(), serDe);
```
Comment (AlexanderSaydakov) [inline src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java:258]: should be k, which is defined above and never used

**Comment by leerho** [comment]
I decided to change the weighted update methods to have a weight type of long.

File: src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java [hunk 3]
```diff
@@ -205,13 +221,64 @@
...
       String v = itr.getQuantile();
       long wt = itr.getWeight();
       printf("%12s%12d\n", v, wt);
+      assertEquals(wt, correct[i++]);
     }
   }
 
+  @Test //set static enablePrinting = true for visual checking
+  public void checkWeightedUpdates1() {
+    int k = 20;
```
[the remark this answers is not in the document]
Comment (leerho) [reply src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java:236]: Fixed

File: src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java [hunk 4]
```diff
@@ -205,13 +221,64 @@
...
+    assertEquals(sk.getN(), weight);
+    sk.update(item, weight);
+    println(sk.toString(true, true));
+    assertEquals(sk.getNumRetained(), 14);
+    assertEquals(sk.getN(), 254);
+  }
+
+  @Test //set static enablePrinting = true for visual checking
+  public void checkWeightedUpdates2() {
+    int k = 20;
```
[the remark this answers is not in the document]
Comment (leerho) [reply src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java:253]: Fixed

File: src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java (hunk 1 above)
Comment (leerho) [reply src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java]: Fixed

File: src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java (hunk 2 above)
Comment (leerho) [reply src/test/java/org/apache/datasketches/kll/KllMiscItemsTest.java]: Fixed

**Review by AlexanderSaydakov — APPROVED** [review]

**Pull request merged**